HW04.Rmd

```diff
@@ -1,126 +1,266 @@
+
+# read in the data
+gun <- read.csv("gun_deaths.csv", header = TRUE, stringsAsFactors = FALSE)
+head(gun)
+
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+library(knitr)
+#year <- gun$year
+#month <- gun$month
+#year_month <- paste(year,month,sep="/")
+#year_month <- table(year_month) 
+#year_month <- as.data.frame(year_month)
+#colnames(year_month) <- c("Year_month", "Gun_Death")
+#kable(year_month, format = "markdown")
+
+month_gun <- table(gun$month)
+month_gun <- as.data.frame(month_gun)
+colnames(month_gun) <- c("month","gun_death")
+kable(month_gun, format = "markdown")
+
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+month_gun$month <- month.name
+#need to make month an ordered factor, so ggplot doesn't do it alphabaticially
+#Factors are used to represent categorical data. Factors can be ordered or unordered and are an important class for statistical analysis and for plotting.
+month_gun$month <- factor(month_gun$month, levels = month_gun$month)
+
+ggplot(month_gun)+
+    geom_bar(aes(x = month, y = gun_death, fill = month), stat = "identity")+
+    labs(x = "Month", y = "Gun death per month", title = "Gun death per month")+
+    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
+        panel.background = element_blank(), axis.line = element_line(colour = "black"), 
+        legend.position = "none")
+
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+
+gun_intent <- gun$intent
+gun_intent <- as.data.frame(table(gun_intent))
+colnames(gun_intent) <- c("intent","death")
+
+ggplot(gun_intent)+
+    geom_bar(aes(x = intent,  y = death, fill = intent), stat = "identity")+
+    labs(x = "Intent", y = "Gun death", title = "Gun death for each intent")+
+    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
+        panel.background = element_blank(), axis.line = element_line(colour = "black"), 
+        legend.position = "none")
+
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+
+gun_age_sex <- gun[,5:6]
+#gun_age_sex <- as.data.frame(table(gun_age_sex))
+#colnames(gun_age_sex) <- c("sex", "age", "death")
+
+#remove the not available line
+gun_age_sex <- subset(gun_age_sex, age != "Not Available")
+#gun_age_sex <- subset(gun_age_sex, death != 0)
```

Comment:
I did not know "!=" to remove certain rows!! I think I could have applied this to my code to improve my efficiency!